x/wasm/client/cli/gov_tx.go

```diff
@@ -0,0 +1,836 @@
+package cli
+
+import (
+	"bytes"
+	"crypto/sha256"
+	"fmt"
+	"net/url"
+	"strconv"
+	"strings"
+
+	"github.com/cosmos/cosmos-sdk/client"
+	"github.com/cosmos/cosmos-sdk/client/flags"
+	"github.com/cosmos/cosmos-sdk/client/tx"
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	"github.com/cosmos/cosmos-sdk/types/address"
+	"github.com/cosmos/cosmos-sdk/version"
+	"github.com/cosmos/cosmos-sdk/x/gov/client/cli"
+	v1 "github.com/cosmos/cosmos-sdk/x/gov/types/v1"
+	"github.com/docker/distribution/reference"
+	"github.com/pkg/errors"
+	"github.com/spf13/cobra"
+	flag "github.com/spf13/pflag"
+
+	"github.com/CosmWasm/wasmd/x/wasm/ioutils"
+	"github.com/CosmWasm/wasmd/x/wasm/types"
+)
+
+func SubmitProposalCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:          "submit-proposal",
+		Short:        "Submit a wasm proposal.",
+		SilenceUsage: true,
+	}
+	cmd.AddCommand(
+		ProposalStoreCodeCmd(),
+		ProposalInstantiateContractCmd(),
+		ProposalInstantiateContract2Cmd(),
+		ProposalStoreAndInstantiateContractCmd(),
+		ProposalMigrateContractCmd(),
+		ProposalExecuteContractCmd(),
+		ProposalSudoContractCmd(),
+		ProposalUpdateContractAdminCmd(),
+		ProposalClearContractAdminCmd(),
+		ProposalPinCodesCmd(),
+		ProposalUnpinCodesCmd(),
+		ProposalUpdateInstantiateConfigCmd(),
+	)
+	return cmd
+}
+
+func ProposalStoreCodeCmd() *cobra.Command {
+	cmd := &cobra.Command{
+		Use:   "wasm-store [wasm file] --title [text] --summary [text] --authority [address]",
+		Short: "Submit a wasm binary proposal",
+		Args:  cobra.ExactArgs(1),
+		RunE: func(cmd *cobra.Command, args []string) error {
+			clientCtx, proposalTitle, summary, deposit, err := getProposalInfo(cmd)
+			if err != nil {
+				return err
+			}
+			authority, err := cmd.Flags().GetString(flagAuthority)
+			if err != nil {
+				return fmt.Errorf("authority: %s", err)
+			}
+
+			if len(authority) == 0 {
+				authority = sdk.AccAddress(address.Module("gov")).String()
```

Comment:
I was expecting that the address would be queried but I guess it is alright to set it for the client already.
While the address can be changed in the keeper(s) on init, it can not be modified on the CLI. I have opened https://github.com/CosmWasm/wasmd/pull/1356 to have a public var instead. Please 👁️